src/components/App.tsx

```diff
@@ -0,0 +1,99 @@
+import { useCallback, useState } from "react";
+
+import { Key } from "./Key";
+import { Button } from "./Button";
+
+import { Api } from "../utils/Api";
+
+type KeyType = {
+  id: string;
+  value: string;
+};
+
+/*
+  Легенда:
+  Необходимо отобразить список ключей (произвольные строки), запрашиваемый с бэкенда.
+  При нажатии кнопки "Сгенерировать ключ" необходимо сгенерировать новый ключ (апи запрос) и отобразить его в списке.
+  Каждый ключ можно использовать. В таком случае он должен изменить свое визуальное отображение и увеличить счетчик использованных ключей.
+  Каждый ключ можно удалить. При удалении использованного ключа, счетчик использованных ключей уменьшается (счетчику важны только неудаленные использованные ключи)
+
+  Для поддержания актуального списка ключей необходимо обновлять список каждые 30 секунд. При получении от апи новых ключей считать их неиспользованными
+ */
+
+export const App = () => {
+  const [keys, setKeys] = useState<any>(null);
```

Comment:
У нас есть тип KeyType, можно тут его указать

```suggestion
  const [keys, setKeys] = useState<KeyType[] | null>(null);
```

Reply:
[]